multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaArchiveStep.java

```diff
@@ -46,23 +48,32 @@ protected StepPhase executeStep(ProcessContext context) {
         String appArchiveId = context.getRequiredVariable(Variables.APP_ARCHIVE_ID);
         getStepLogger().debug(Messages.MTA_ARCHIVE_ID_0_MESSAGE, appArchiveId);
         processApplicationArchive(context, appArchiveId);
+
         setMtaIdForProcess(context);
         acquireOperationLock(context);
         getStepLogger().debug(Messages.MTA_ARCHIVE_PROCESSED);
         return StepPhase.DONE;
     }
 
     private void processApplicationArchive(ProcessContext context, String appArchiveId) {
-        List<ArchiveEntryWithStreamPositions> archiveEntriesWithStreamPositions = archiveEntryStreamWithStreamPositionsDeterminer.determineArchiveEntries(context.getRequiredVariable(Variables.SPACE_GUID),
-                                                                                                                                                          appArchiveId);
+        List<ArchiveEntryWithStreamPositions> archiveEntriesWithStreamPositions = archiveEntryStreamWithStreamPositionsDeterminer.determineArchiveEntries(
+            context.getRequiredVariable(Variables.SPACE_GUID), appArchiveId);
         context.setVariable(Variables.ARCHIVE_ENTRIES_POSITIONS, archiveEntriesWithStreamPositions);
         MtaArchiveHelper helper = createMtaArchiveHelperFromManifest(context, appArchiveId, archiveEntriesWithStreamPositions);
+
+        DeploymentDescriptor deploymentDescriptor = extractDeploymentDescriptor(context, appArchiveId, archiveEntriesWithStreamPositions);
+
+        if (context.getVariable(Variables.SHOULD_BACKUP_PREVIOUS_VERSION)) {
+            MtaArchiveContentResolver contentResolver = getMtaArchiveContentResolver(context, deploymentDescriptor);
+            contentResolver.resolveMtaArchiveFilesInDescriptor(appArchiveId, helper);
```

Comment:
pass the descriptor to the method and pass the helper in the constructor, as the descriptor will be actually changed by the method invocation

Reply:
Done